cmd/operator/cmd.go

```diff
@@ -88,6 +91,35 @@ func runLocally(kubeCfg *rest.Config) error {
 }
 
 func runOperator(kubeCfg *rest.Config, namespace string, isOLM bool) error {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(kubeCfg)
+	if err != nil {
+		return err
+	}
+
+	groupVersion := schema.GroupVersion{
+		Group:   "dynatrace.com",
+		Version: expectedVersion,
+	}
+
+	resourceList, err := discoveryClient.ServerResourcesForGroupVersion(groupVersion.String())
+	if err != nil {
+		return errors.WithMessagef(err, "required DynaKube CRD version %s not found", groupVersion.String())
+	}
+
+	var versionFound bool
+
+	for _, resource := range resourceList.APIResources {
+		if resource.Kind == "DynaKube" {
+			versionFound = true
+
+			break
+		}
+	}
+
+	if !versionFound {
+		return errors.New("DynaKube kind not found in group version")
+	}
+
```

Comment:
```suggestion
	groupKind := schema.GroupKind{
		Group: v1beta4.GroupVersion.Group,
		Kind: "DynaKube", // maybe use reflect to get the name of the DynaKube stuct, shrug
	}

	_, err = operatorManager.GetRESTMapper().RESTMapping(groupKind, v1beta4.GroupVersion.Version)
	if err != nil {
		log.Info("missing CRD")
		return err
	}
```

didn't test it, also this only considers the `DynaKube`

you could also do similar with `EdgeConnect`

Reply:
i'm about to propose same, i like this idea, bc when we bump version we will update this file easy, rather than change string.

Also you can revisit what we do here https://github.com/dynatrace/dynatrace-operator/blob/02ee3154eb669436c4df8f49aaa84486d6f8ed1f/cmd/support_archive/resources.go#L142-L165

Reply:
+1 for EdgeConnect

Reply:
should be:

```
	groupVersion := schema.GroupVersion{
		Group:   v1beta4.GroupVersion.Group,
		Version: v1beta4.GroupVersion.Version,
	}
```

Reply:
regarding name +1 https://pkg.go.dev/reflect#TypeOf

Reply:
Done https://github.com/Dynatrace/dynatrace-operator/pull/4763/commits/15e868540889ad0ee13ac834d9dd72d662226b7c